assignments/hw4/features.py

```diff
@@ -0,0 +1,54 @@
+"""Module for determining feature types.
+"""
+
+import sys
+from enum import Enum
+
+import data
+from pandas import Series
+
+
+class FeatureType(Enum):
+    CONTINUOUS = 0
+    CATEGORICAL = 1
+    BOOLEAN = 2
+
+
+def determine_response_type(feature: Series) -> FeatureType:
+    """Continuous or boolean."""
+    # If there are only two values in response,
+    # we know it's boolean.
+    if len(feature.unique()) == 2:
+        return FeatureType.BOOLEAN
+    # Otherwise, we can assume its continuous
+    return FeatureType.CONTINUOUS
+
+
+def determine_predictor_type(
+    feature: Series,
+    threshold: float = 0.05,
+) -> FeatureType:
+    """Continuous or categorical."""
+    # Assume category or object dtypes are categorical
+    if feature.dtype.name in ["category", "object"]:
+        return FeatureType.CATEGORICAL
+    # Categories encoded in integers are tricky...
+    # Found an answer in Stack Overflow Q #35826912
+    # that gave a heuristic for guessing this.
+    if float(feature.nunique()) / feature.count() < threshold:
+        return FeatureType.CATEGORICAL
```

Comment:
This is a clever idea, and I would be interested in finding out how the threshold changes the output with datasets of varying sizes.